tools/database/gen_db/main.go

```diff
@@ -468,6 +500,47 @@ func InferPhase(item *proto.UIItem) int32 {
 	return 0
 }
 
+func inferCelestialItemSource(item *proto.UIItem) []*proto.UIItemSource {
+	if item.Phase <= 2 {
+		sources := make([]*proto.UIItemSource, 0, len(item.Sources)+1)
+		// Make sure to always add the SoldBy source first so it shows up first in the UI since we pick the first index
+		// but we still need the other sources to add things like Sha-Touched gems
+		sources = append(sources, &proto.UIItemSource{
+			Source: &proto.UIItemSource_SoldBy{SoldBy: &proto.SoldBySource{
+				NpcId:   248108,
+				NpcName: "Avatar of the August Celestials",
+			}},
+		})
+		sources = append(sources, item.Sources...)
+		return sources
+	}
+	return item.Sources
+}
+func inferFlexibleRaidItemSource(item *proto.UIItem) []*proto.UIItemSource {
+	sources := item.Sources
+	hasSources := len(item.Sources) > 0
+	if hasSources {
+		for _, source := range sources {
+			if drop := source.GetDrop(); drop != nil {
+				// Flex raid has no difficulty index so we need to infer it from name
+				if drop.Difficulty == proto.DungeonDifficulty_DifficultyUnknown {
+					drop.Difficulty = proto.DungeonDifficulty_DifficultyRaidFlex
+				}
+			}
+		}
+	} else {
+		// Some Flex items don't have a drop source listed,
+		// so just add the difficulty for filtering
+		sources = append(sources, &proto.UIItemSource{
+			Source: &proto.UIItemSource_Drop{Drop: &proto.DropSource{
+				Difficulty: proto.DungeonDifficulty_DifficultyRaidFlex,
+			}},
+		})
+	}
+
+	return sources
+}
```

Comment:
`main.go` is getting to be way too huge at this point. Can we move the `inferPhase()`, `inferCelestialItemSource()`, and `inferFlexibleRaidItemSource()` functions to a new `item_source_utils.go` file?